forch/port_state_manager.py

```diff
@@ -188,6 +192,28 @@ def handle_static_device_behavior(self, mac, device_behavior):
             self._auto_sequester[mac_lower] = auto_sequester
             if device_behavior.segment:
                 self.handle_device_behavior(mac_lower, device_behavior, static=True)
+            sequester_scheduled = self._scheduled_sequester_timer.get(mac_lower)
+            if device_behavior.scheduled_sequestering_timestamp and not sequester_scheduled:
+                try:
+                    parsed = dateutil.parser.parse(device_behavior.scheduled_sequestering_timestamp)
+                except dateutil.parser.ParserError:
+                    self._logger.error("Failed to parse scheduled sequestering timestamp: %s.",
+                                       device_behavior.scheduled_sequestering_timestamp)
+                    return
+                local_tz = dateutil.tz.tzlocal()
+                if not parsed.tzinfo:
+                    parsed = parsed.replace(tzinfo=local_tz)
+                if parsed >= datetime.now(local_tz):
+                    time_diff = parsed - datetime.now(local_tz)
+
+                    def handler():
+                        self._handle_scheduled_sequstering(mac_lower)
```

Comment:
did you think through the multithreaded processing part here, to make sure the appropriate locks are in place when multiple threads are doing whatever?

Reply:
The multiprocess is contained in the device report client so it doesn't affect anything here; the creation of the timer is done with a lock so each device should only have 1 timer active at any time. It does look like I ignore any changes to schedule time change, so I'll make a change to update this.